### PR TITLE
Local push notifications

### LINKED ISSUE #224: Push Notifications
Not sure how feasible this would be with the API, but have notifications received via push. Even better, customize the repos/types of notifications you would like to recieve pushes for.

### LINKED ISSUE #1888: Not receiving Push Notifications
Wanted to ask first under what circumstances you get a push notifications. I have been mentioned in PRs and issues but I have never received push notifications.

<details>
<summary>Bug Report Dump (Auto-generated)</summary>
<pre>
Version 1.21.0 (1528145114)
Device: iPhone X (iOS 11.4)
TestFlight: false
</pre>
</details>

### LINKED ISSUE #1928: Push Notifications with local push?
The biggest ask I keep seeing is push notifications, however I _really_ don't want to setup a server to send APNs for multiple reasons: I don't want to build/maintain it, and I also don't want to store people's info off of their device (esp a token!).

I've hacked push notifications before using local push. We already have a background job to update the app badge. We could update the local cache with what has been "pushed" before, and for new notifications just queue a local push notification instead.

What do you think? Should we try it?

### PR DESCRIPTION
**Pull request by rnystrom**
- Use FMDB to store seen notifications
- Update db when fetching
- Trim db to latest 1000 notifications
- If bg fetch and push enabled, show local pushes when new ones arrive

Probably more improvements we can make down the road, but it works!

simulator screen shot - iphone x - 2018-08-26 at 22 40 09

Fixes #1928, fixes #1888, fixes #224

### CODE REVIEW

**Comment by rnystrom** [comment]
TODO

- [ ] Clear notifications?
- [ ] Tapping on notification should navigate to item
  - This might be tricky w/out #2105 finished

File: Classes/Systems/BadgeNotifications.swift [hunk 1]
```diff
@@ -13,52 +13,61 @@ import GitHubSession
...
+            defaults.set(newValue, forKey: userKey)
         }
     }
 
-    enum State {
-        case initial
-        case denied
-        case disabled
-        case enabled
+    private static let notificationKey = "com.freetime.BadgeNotifications.notifications-enabled"
```
Comment (jdisho) [inline Classes/Systems/BadgeNotifications.swift]: Why not in a "Constants" file?

Comment (rnystrom) [reply Classes/Systems/BadgeNotifications.swift]: These aren’t used anywhere else

**Comment by gingerbeardman** [comment]
Exciting!

**Comment by rnystrom** [comment]
Definitely some work to do, but its a good first start!

**Comment by mesqueeb** [comment]
Hey guys!! Is this update already live on the iOS store?

**Comment by Huddie** [comment]
@mesqueeb Not yet

**Comment by mesqueeb** [comment]
@Huddie I can't wait!!! ♡

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by j-f1** [comment]
@mesqueeb There’s a TestFlight sign up link in the README where you can get the beta version.

**Pull request merged**